.github/workflows/reusable-workflow.yaml

```diff
@@ -75,55 +75,74 @@ jobs:
       - uses: 'google-github-actions/setup-gcloud@v2'
         if: success() || failure()
       - name: Run Tests
+        continue-on-error: true
         if: success() || failure()
         env:
           # Most used according to https://gs.statcounter.com/android-version-market-share/mobile-tablet/worldwide
-          DEFAULT_API_VERSION: "34" 
+          PR_API_VERSION: "34" 
           FULL_API_RANGE: "28 29 30 31 32 33 34 35"
           IS_PR: ${{ inputs.is_pr }}
         run: |
-          DEVICES=""
+          LEVELS_TO_TEST=$FULL_API_RANGE
+
           if $IS_PR ; then
-            DEVICES="--device model=MediumPhone.arm,version=${DEFAULT_API_VERSION},locale=en,orientation=portrait "
-          else
-            for LEVEL in $FULL_API_RANGE
-            do
-                DEVICES+="--device model=MediumPhone.arm,version=${LEVEL},locale=en,orientation=portrait "
-            done
+            LEVELS_TO_TEST=$PR_API_VERSION
           fi
 
-          COMMAND="gcloud firebase test android run "
-          COMMAND+="--project mobile-apps-firebase-test "
-          COMMAND+="--type instrumentation "
-          COMMAND+="--app \"native/NativeSampleApps/RestExplorer/build/outputs/apk/debug/RestExplorer-debug.apk\" "
-          COMMAND+="--test=libs/${{ inputs.lib }}/build/outputs/apk/androidTest/debug/${{ inputs.lib }}-debug-androidTest.apk "
-          COMMAND+="${DEVICES}"
-          COMMAND+="--environment-variables coverage=true,coverageFile=\"/sdcard/coverage.ec\" "
-          COMMAND+="--directories-to-pull=/sdcard  "
-          COMMAND+="--results-dir=${{ inputs.lib }}-${{github.run_number}} "
-          COMMAND+="--results-history-name=${{ inputs.lib }} "
-          COMMAND+="--timeout=20m --no-auto-google-login --no-record-video --no-performance-metrics --num-flaky-test-attempts=1"
+          mkdir firebase_results
+          for LEVEL in $LEVELS_TO_TEST
+            do
+              GCLOUD_RESULTS_DIR=${{ inputs.lib }}-api-${LEVEL}-build-${{github.run_number}}
 
-          eval "$COMMAND"
+              gcloud firebase test android run \
+                --project mobile-apps-firebase-test \
+                --type instrumentation \
+                --app "native/NativeSampleApps/RestExplorer/build/outputs/apk/debug/RestExplorer-debug.apk" \
+                --test=libs/${{ inputs.lib }}/build/outputs/apk/androidTest/debug/${{ inputs.lib }}-debug-androidTest.apk \
+                --device model=MediumPhone.arm,version=${LEVEL},locale=en,orientation=portrait \
+                --environment-variables coverage=true,coverageFile="/sdcard/coverage.ec" \
+                --directories-to-pull=/sdcard \
+                --results-dir=${GCLOUD_RESULTS_DIR} \
+                --results-history-name=${{ inputs.lib }} \
+                --timeout=20m --no-auto-google-login --no-record-video --no-performance-metrics --num-flaky-test-attempts=1 || true
```

Comment:
The `|| true` is a little trick to ensure the test run does not return a non-zero value to the shell, since that will cause the entire step to end before we have looped through all API levels.